event-store-plans.md

```diff
@@ -0,0 +1,174 @@
+# Event Store Improvements for V7
+
+This document is brainstorming for improvements to the Event Store functionality in Marten V7. This does not include features meant for
+"CritterStackPro" products, but does include some necessary baseline infrastructure for planned "CritterStackPro" features.
+
+In some cases, this document outlines "poor man's" subset strategies for features that we will build in a more robust (and inevitably more complicated) way 
+in "CritterStackPro"
+
+The goal is to start a discussion about what is and isn't going to be in scope for V7. When this document is "complete", the output is a new set
+of GitHub issues marked for the V7 release. 
+
+
+## Raise events from asynchronous projection updates
+
+I'm going to allow Oskar address this one:-)
+
+From my side, I think the `ISessionListener` approach we have today is weak and should be a little more "outbox'd" such that it only fires off when
```

Comment:
I think that there are actually two things here. From what I see, async session listeners are usually used when people want to publish the changes to read models outside the world, so as I described in my blog article: https://event-driven.io/en/publishing_read_model_changes_from_marten/.

Some people are fine with just forwarding the change to the read model; this scenario is fine for building a local copy of data between modules.

The other scenario is enriching events. For cases like "shopping cart confirmation", external processes can be triggered (e.g. "issuing invoice"). Internal events should be granular so they won't contain all the information. Some people want to enrich it either based on the updated read model or with an aggregating stream and send new events (so as I explained in https://event-driven.io/en/internal_external_events/).

Those two scenarios don't necessarily have to be implemented the same way. For the read model propagation, we may use logical replication, so use it as I prototyped here: https://github.com/oskardudycz/PostgresOutboxPatternWithCDC.NET.

For events enrichment, I think that we allow appending new events in the projections/subscriptions, but then defer calling external systems with projections based on the enriched events.

I also believe that having support for that will enable lightweight saga processing.

The biggest question here is what will be part of Wolverine and what will be part of Marten. I'm voting that the stuff I mentioned should be part of vanilla Marten. The logical replication part could be a plugin (could be part of CritterStackPro).